### PR TITLE
Feature/opt in properties

### PR DESCRIPTION
**Pull request by PawelPawelec-RDX**

### CODE REVIEW

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Services/TransactionQuerier.cs [hunk 1]
```diff
@@ -219,22 +212,43 @@ private GatewayModel.CommittedTransactionInfo MapToGatewayAccountTransaction(Led
...
+            rawPayload: includeRawPayload == true ? ult.RawPayload.ToHex() : null,
+            receipt: new GatewayModel.TransactionReceipt
+            {
+                ErrorMessage = ult.EngineReceipt.ErrorMessage,
+                Status = MapTransactionStatus(ult.EngineReceipt.Status),
+                Items = new JRaw(ult.EngineReceipt.Items),
+                FeeSummary = includeFeeSummary == true ? new JRaw(ult.EngineReceipt.FeeSummary) : null,
+                NextEpoch = ult.EngineReceipt.NextEpoch != null ? new JRaw(ult.EngineReceipt.NextEpoch) : null,
+                StateUpdates = includeStateChanges == true ? new JRaw(ult.EngineReceipt.StateUpdates) : null,
+                Events = includeEvents == true ? new JRaw(ult.EngineReceipt.Events) : null,
```
Comment (KrzysztofLabus-RDX) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/Services/TransactionQuerier.cs:241]: Ideally we should not even fetch them from DB but guess this should be good-enough.,

File: src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml [hunk 2]
```diff
@@ -1359,8 +1378,19 @@ components:
           properties:
             intent_hash_hex:
               $ref: "#/components/schemas/HashHexString"
+            opt_in_properties:
+              $ref: "#/components/schemas/TransactionCommittedDetailsOptInProperties"
       example:
         intent_hash_hex: "<transaction-intent-hash>"
+    TransactionCommittedDetailsOptInProperties:
```
[the remark this answers is not in the document]
Comment (PawelPawelec-RDX) [reply src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml:1385]: well I checked that and it looked resonable for both ts and .net

```        public enum OptInPropertiesEnum
        {
            /// <summary>
            /// Enum RawPayload for value: raw_payload
            /// </summary>
            [EnumMember(Value = "raw_payload")]
            RawPayload = 1,

            /// <summary>
            /// Enum ReceiptStateChanges for value: receipt.state_changes
            /// </summary>
            [EnumMember(Value = "receipt.state_changes")]
            ReceiptStateChanges = 2,

            /// <summary>
            /// Enum ReceiptFeeSummary for value: receipt.fee_summary
            /// </summary>
            [EnumMember(Value = "receipt.fee_summary")]
            ReceiptFeeSummary = 3,

            /// <summary>
            /// Enum ReceiptEvents for value: receipt.events
            /// </summary>
            [EnumMember(Value = "receipt.events")]
            ReceiptEvents = 4
        }```

```export const StateEntityDetailsRequestOptInPropertiesEnum = {
    AncestorIdentities: 'ancestor_identities',
    ComponentRoyaltyVaultBalance: 'component.royalty_vault_balance',
    PackageRoyaltyVaultBalance: 'package.royalty_vault_balance'
} as const;```

File: src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml [hunk 3]
```diff
@@ -1376,14 +1406,13 @@ components:
     TransactionCommittedDetailsResponseDetails:
       type: object
       required:
-        - raw_hex
         - receipt
         - referenced_global_entities
       properties:
-        raw_hex:
+        raw_payload:
```
[the remark this answers is not in the document]
Comment (PawelPawelec-RDX) [reply src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml:1412]: I'm happy with old name, will revert just thought we want to change it based on our confluence document.

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Services/TransactionQuerier.cs (hunk 1 above)
Comment (PawelPawelec-RDX) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/Services/TransactionQuerier.cs:241]: I can do that, but that would mean replacing that entity framework query with raw sql with dapper.
If you feel it's right time for that, happy to address that.

Comment (KrzysztofLabus-RDX) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/Services/TransactionQuerier.cs:241]: I don't think we need to invest that much into this area at the moment.

File: src/RadixDlt.NetworkGateway.GatewayApi/Services/IEntityStateQuerier.cs [hunk 4]
```diff
@@ -76,7 +76,7 @@ public sealed record PageRequest(GlobalAddress Address, int Offset, int Limit);
 
     public sealed record ResourceVaultsPageRequest(GlobalAddress Address, GlobalAddress ResourceAddress, int Offset, int Limit);
 
-    Task<GatewayModel.StateEntityDetailsResponseItem> EntityDetailsItem(GlobalAddress address, bool aggregatePerVault, GatewayModel.LedgerState ledgerState, CancellationToken token = default);
+    Task<GatewayModel.StateEntityDetailsResponseItem> EntityDetailsItem(GlobalAddress address, bool aggregatePerVault, GatewayModel.StateEntityDetailsOptInProperties? optInProperties, GatewayModel.LedgerState ledgerState, CancellationToken token = default);
```
Comment (KrzysztofLabus-RDX) [inline src/RadixDlt.NetworkGateway.GatewayApi/Services/IEntityStateQuerier.cs:79]: Make it mandatory property and use something like `request.OptIns ?? XxxOptIns.Default` in controller?

File: src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml [hunk 5]
```diff
@@ -1444,6 +1482,8 @@ components:
           required:
             - addresses
           properties:
+            opt_in_properties:
```
Comment (KrzysztofLabus-RDX) [inline src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml:1485]: OptIns vs OptInProperties... Somehow former looks better to me but I have no strong opinion on that.

**Review by KrzysztofLabus-RDX — APPROVED** [review]

**Pull request merged**